addon/-private/data-view/radar/index.js

```diff
@@ -63,30 +128,156 @@ export default class Radar {
+      } else if (itemDelta > 0) {
+        // Scrolling down
+        let movedComponents = orderedComponents.splice(0, offsetAmount);
+
+        orderedComponents[0].hasBeenMeasured = false;
+        orderedComponents.push(...movedComponents);
+
+        VirtualComponent.moveComponents(_itemContainer, movedComponents[0], movedComponents[movedComponents.length - 1], false);
+      }
+    }
+
+    for (let i = 0, itemIndex = firstItemIndex; itemIndex <= lastItemIndex; i++, itemIndex++) {
+      orderedComponents[i].recycle(items[itemIndex], itemIndex);
+    }
+
+    _itemContainer.style.paddingTop = `${totalBefore}px`;
+    _itemContainer.style.paddingBottom = `${totalAfter}px`;
+
+    this._prevFirstItemIndex = firstItemIndex;
+  }
+
+  /*
+   * Sets up _virtualComponents, which is meant to be a static pool of components that we render to.
+   * In order to decrease the time spent rendering and diffing, we pull the {{each}} out of the DOM
+   * and only replace the content of _virtualComponents which are removed/added.
+   *
+   * For instance, if we start with the following and scroll down, items 2 and 3 do not need to be
+   * rerendered, only item 1 needs to be removed and only item 4 needs to be added. So we replace
+   * item 1 with item 4, and then manually move the DOM:
+   *
+   *   1                        4                         2
+   *   2 -> replace 1 with 4 -> 2 -> manually move DOM -> 3
+   *   3                        3                         4
+   *
+   * However, _virtualComponents is still out of order. Rather than keep track of the state of
+   * things in _virtualComponents, we track the visually ordered components in the
+   * _orderedComponents array. This is possible because all of our operations are relatively simple,
+   * popping some number of components off one end and pushing them onto the other.
+   *
+   * @private
+   */
+
+  _updateVirtualComponentPool() {
+    const {
+      _scrollContainerHeight,
+      bufferSize,
+      minHeight,
+      virtualComponents,
+      orderedComponents,
+      totalItems
+    } = this;
+
+    // The total number of components is determined by the minimum number required to span the
+    // container with its buffers. Combined with the above rendering strategy this fairly
+    // performant, even if mean item size is above the minimum.
+    const totalHeight = _scrollContainerHeight + (_scrollContainerHeight * bufferSize * 2);
+    const totalComponents = Math.min(totalItems, Math.ceil(totalHeight / minHeight) + 1);
+    const delta = totalComponents - virtualComponents.get('length');
+
+    if (delta) {
+      if (delta > 0) {
+        for (let i = 0; i < delta; i++) {
+          let component = new VirtualComponent(this.token);
+          set(component, 'content', {});
+
+          virtualComponents.pushObject(component);
+        }
+      } else {
+        for (let i = virtualComponents.length; i > delta; i--) {
```

Comment:
Good catch, fixed that. This is the logic I was talking about for resizing by the way, it's roughly there but it doesn't really work yet.